src/Desktop/UIAutomation/CustomObjects/Config.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Newtonsoft.Json;
+using System;
+using System.IO;
+using System.Linq;
+
+namespace Axe.Windows.Desktop.UIAutomation.CustomObjects
+{
+    class Config
+    {
+        [JsonProperty("properties")]
+        public CustomProperty[] Properties { get; set; }
+
+        public static Config ReadFromText(string text)
+        {
+            Config conf = JsonConvert.DeserializeObject<Config>(text);
+            conf.Validate();
+            // We made it here, so config must be valid.
+            return conf;
+        }
+
+        public static Config ReadFromFile(string path) { return Config.ReadFromText(File.ReadAllText(path)); }
+
+        private void Validate()
+        {
+            if (Properties == null || !Properties.Any()) throw new ArgumentException("Empty or missing definition of custom properties.");
```

Comment:
As designed, ArgumentException is intended for use against a parameter, and the parameter name should be in the exception. In this case, since we don't have a parameter, let's throw [InvalidDataException](https://docs.microsoft.com/en-us/dotnet/api/system.io.invaliddataexception?view=net-5.0).

Same logic applies to CustomProperty.Validate() and CustomProperty.ValidateType(), neither of which has a parameter